quic/s2n-quic-core/src/datagram/default.rs

```diff
@@ -78,10 +232,178 @@ impl Builder {
         self.queue_capacity = capacity;
         self
     }
+
+    /// Gives the default sender relevant connection info
+    pub fn with_connection_info(mut self, connection_info: ConnectionInfo) -> Self {
+        self.connection_info = connection_info;
+        self
+    }
+
     /// Builds the datagram sender into a provider
     pub fn build(self) -> Result<DefaultSender, core::convert::Infallible> {
         Ok(DefaultSender {
             queue: VecDeque::with_capacity(self.queue_capacity),
+            capacity: self.queue_capacity,
+            max_datagram_payload: self.connection_info.max_datagram_payload,
+            max_packet_space: 0,
+            min_packet_space: 0,
+            smoothed_packet_size: 0.0,
+            waker: None,
         })
     }
 }
+
+#[test]
```

Comment:
I would put all of these in a module
```suggestion
#[cfg(test)]
mod tests {
    use super::*;
    
    #[test]
```